### ISSUE TITLE
Endgame lab was generated with no finales

### ISSUE THREAD

**Issue by iillyyaa2033** [cataclysmbnteam/Cataclysm-BN#37]
- Game version: commit 5f652ac
- Operating system: linux x64
- Tiles or curses: curses
- Mods active: [see here]https://user-images.githubusercontent.com/17745136/90317040-43aba200-df2f-11ea-84dd-0f806eb89a19.png)

### Actual behavior
![image](https://user-images.githubusercontent.com/17745136/90317467-77d49200-df32-11ea-8631-29380f1f7e81.png)

### Steps to reproduce the behavior
Same as in #36 

Also there's no debug.log in `config` folder so I cannot provide logs.

**Comment by Coolthulhu** [comment]
Now this was a hard to test bug, because I had to test whole overmap generation at once rather than isolating bad parts.
The bug was caused by an optimization that prevents generating z-level -3 unless hardcoded overmapgen on z-level -2 requests generating another z-level.
Specials don't trigger this request with the json-coded part.

So the labs generated properly on any overmap that has a hardcoded special that goes to z-level -3. A regular lab, a slime pit, a mine. Once it reached -3, the central lab could work from there, but if nothing hardcoded reached that, the lab would not generate properly.
